spec/beaker-hostgenerator/generator_spec.rb

```diff
@@ -98,12 +57,20 @@ module BeakerHostGenerator
             fixture_hash['environment_variables'].each do |key, value|
               ENV[key] = value
             end
+
             cli = BeakerHostGenerator::CLI.new(arguments)
-            test_hash = YAML.load(cli.execute)
+            if fixture_hash['expected_exception']
+              # Turn fully-qualified classname string into an actual Class object.
+              expected_class = eval fixture_hash['expected_exception']
```

Comment:
Yeah, the rails implementation is even hairier: https://github.com/rails/rails/blob/861b70e92f4a1fc0e465ffcf2ee62680519c8f6f/activesupport/lib/active_support/inflector/methods.rb#L232-L279

Eval is a pretty loaded gun. One thing we might do here is do a regex sanity check that there's nothing in the string but what could only be a class name.  Use `\A` and `\z` instead of `^` and `$` for this (due to multi-line security problems).

So maybe something like (double-check my regex, please):

``` ruby
  class_name = fixture_hash['expected_exception']
  expected_class = constantize(class_name) if class_name
end

def constantize(string)
  raise "Invalid class name '#{string}'" unless valid_class_name?(string)
  eval string
end

def valid_class_name?(string)
  string =~ %r{\A(?:[A-Z][A-Za-z0-9_]*)(?:::[A-Z][A-Za-z0-9_]*)*\z}
end
```